front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/index.tsx

```diff
@@ -316,11 +292,14 @@ const HeroBannerForm = ({ intl: { formatMessage } }: InjectedIntlProps) => {
         <Warning>
           <FormattedMessage {...messages.heroBannerInfoBar} />
         </Warning>
-        <Outlet
-          id="app.containers.Admin.settings.customize.headerSectionStart"
-          homepageSettings={localHomepageSettings}
-          handleOnChange={handleSettingOnChange}
-        />
+
+        {type === 'homePage' && (
+          <Outlet
+            id="app.containers.Admin.settings.customize.headerSectionStart"
+            bannerLayout={localSettings.banner_layout ?? 'two_column_layout'}
+            handleOnChange={updateValueInLocalState}
+          />
+        )}
```

Comment:
Outlets can't be rendered conditionally. Condition needs to be in the module that renders something in the outlet.

Reply:
I remember the rule but this is for a different case, the outlets are only applicable when the form represents a HomePage. It's effectively two forms, one with the outlets always rendered (`homePage`), one with them never rendered (`customPage`)

Reply:
Outlets should never be rendered conditionally. The condition needs to be in a different place. :-D But maybe it's solved with the extra form in between.

Reply:
Well, here's the use case, and why I think it's a different case:

- Custom Page: you can't set the layout (yet) or the CTA button, so the outlets never render. This has nothing to do with the `customizable_homepage_banner` feature flag
- HomePage: you can set both on this page, so the outlet code always renders, and then render further based on the user's `customizable_homepage_banner` feature flag. We can't include this outlet when the form is for a CustomPage because it would render based on the feature flag, not based on the properties of the CustomPage

In other words, to me it's more like two separate components:
```<CustomPageHeroBannerForm>
<Outlet id="HeaderSectionStart" /> (outlet always)
</CustomPageHeroBannerForm>

<HomePageHeroBannerForm>
{children} (outlet never)
</HomePageHeroBannerForm>```

If that's not right then we'll probably need to do a bigger refactor, especially since I think the CTA/layout are now available for all users and could just be moved into the main app, but I wanted to get the feature done quickly

Reply:
I don't think I get it yet. I'm going to dive into the code after my break.

In the meanwhile. What do you mean with

> Custom Page: you can't set the layout (yet) or the CTA button, so the outlets never render. This has nothing to do with the customizable_homepage_banner feature flag

In the Figma design, it has the layout and CTA button setting for custom pages, right? 

I am wondering why we don't have a signed in/out button distinction for the custom pages though 🤔 I assumed the settings would be exactly the same as for the home page.

Reply:
My mistake, thought banner layout and CTA settings for custom pages were in a later iteration. If layout and CTA settings are relevant for both types of page then the outlets can always be rendered, but due to the feature flags on the outlet their display would then depend on the user's enabled feature flags and not the type of page they're on. We could move them out of the module into the main app but that's a decent amount of refactoring of `front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/index.tsx` and all related files, to move them into the main app. 

Alternatively if these features are actually available for all users now, we could just remove the feature flag, requiring no refactoring/moving code, and create a ticket to follow up on it later. Would have to update CTASettings to only show one field on CustomPage and two on HomePage but that should be simple.

```
  outlets: {
    'app.containers.Admin.settings.customize.headerSectionStart': (props) => {
      return (
          <LayoutSetting {...props} />
      );
    },
    'app.containers.Admin.settings.customize.headerSectionEnd': (props) => {
      return (
          <CTASettings {...props} />
      );
    },
```

Happy to do either but will defer to you in terms of timing and effort in finishing the feature

I think we discussed the signed in/out buttons for custom pages it and thought that a signed in/signed out distinction did not make sense for pages other than the homepage, as well as leaving it out for simplicity in creating the new feature

@brentguf

Reply:
`front/app/modules/commercial/customizable_homepage_banner/admin/CTASettings/index.tsx` has different values (signed in/out) so it can't be reused. Besides that, the functionality will stay in the module for the homepage, even if it's available for everyone.

We can reuse components though for the custom page. Even if the homepage part is in a module.

Not sure what your problem currently is. Let's do a call.